drivers/gpio/gpio_xlnx_ps.c

```diff
@@ -41,7 +44,31 @@ static const struct gpio_driver_api gpio_xlnx_ps_default_apis;
  */
 static int gpio_xlnx_ps_init(const struct device *dev)
 {
-	const struct gpio_xlnx_ps_dev_cfg *dev_conf = dev->config;
+	const struct gpio_xlnx_ps_dev_cfg *dev_conf = DEV_CFG(dev);
+	struct gpio_xlnx_ps_dev_data *dev_data = DEV_DATA(dev);
+	uint32_t bank;
+
+	/* Perform the actual memory map operation in the parent device */
+	DEVICE_MMIO_NAMED_MAP(dev, reg_base, K_MEM_CACHE_NONE);
+	dev_data->base = DEVICE_MMIO_NAMED_GET(dev, reg_base);
+
+	__ASSERT(dev_data->base != 0, "%s map register space failed", dev->name);
+	if (dev_data->base == 0) {
```

Comment:
No need for both an ASSERT and a run-time check.

Reply:
OK, the run-time check is now gone.